### PR TITLE
Bugfix: Configure scheduled tasks to expire after some time

### LINKED ISSUE #2604: [BUG] Duplicate tasks pile up if the task-queue is offline
### Discussed in https://github.com/paperless-ngx/paperless-ngx/discussions/2593

<div type='discussions-op-text'>

<sup>Originally posted by **hashworks** February  3, 2023</sup>
### Description

Recent versions added a new service to run, paperless-task-queue, otherwise known as the Celery Workers.

They handle tasks that get queued in the redis DB. This works fine for normal system operation where all required processes are running fine.

If the task-queue service goes down for a while, the webserver, the consumer and the scheduler service don't mind at all - they will happily queue new tasks in the redis DB since their job is done. This is a problem when one restarts the task-queue service days later:

```
[2023-02-03 13:08:34,827] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[8bcc8e12-dd69-4c91-826b-1f3388f4b1ec] received
[2023-02-03 13:08:35,220] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[c7c52433-7a1a-4dc6-8895-92ff5e90e152] received
[2023-02-03 13:09:16,771] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[e839b305-63a3-4617-99a5-fc83985194ab] succeeded in 41.953297510015545s: 'No new documents were added.'
[2023-02-03 13:09:16,833] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[d1e5fe78-9b16-4f93-b4eb-accfe6031d8e] succeeded in 41.921987653011456s: 'No new documents were added.'
[2023-02-03 13:09:16,839] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[6c80a09b-0ab3-470c-8305-712193ec902b] succeeded in 42.04609970000456s: 'No new documents were added.'
[2023-02-03 13:09:17,062] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[3bf543ea-189e-406e-b45c-9c2dbfa4037f] received
[2023-02-03 13:09:17,063] [INFO] [celery.worker.strategy] Task documents.tasks.index_optimize[689bf96f-7077-47bb-a9c7-76c3362b1279] received
[2023-02-03 13:09:17,107] [INFO] [celery.app.trace] Task documents.tasks.train_classifier[cab4d3ff-2809-4570-85f9-3339f33a645b] succeeded in 0.02393642399692908s: None
[2023-02-03 13:09:17,109] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[d2ae11b2-0c9c-4920-abe2-599f6843eda6] received
[2023-02-03 13:09:19,266] [INFO] [celery.worker.strategy] Task documents.tasks.train_classifier[49f73466-33e0-4d76-8956-a1ac02ab3e19] received
```

The task-queue will then try to handle all queued up tasks, including classifiers and mail account checks that are normally only run every few minutes. This results in multiple hours of mail checking where no new documents are handled.

Tasks that are queued by a cron job should have a reasonable timeout. If they are added every 10 minutes, they should be discarded if they are older than that.

### Steps to reproduce

1. Use tasks that are added by a cron job, like mail checking
2. Stop the celery workers that handle the task queue
3. Wait a few days
4. Restart celery workers
5. See that one needs to wait a few hours for the celery workers to handle all piled up tasks

### Paperless-ngx version

1.12.2

### Host OS

Arch Linux

### Installation method

Bare metal

### Other

A related issue is #2549. It was closed by recommending a redis reset, but I think a task timeout would be a better solution.</div>

### PR DESCRIPTION
**Pull request by stumpylog**
## Proposed change

This configures the scheduled tasks to expire based on the default timing.  This means if the worker goes away for some reason, it won't come back to a deluge of waiting tasks, all of which are doing the same thing.

I chose to expire tasks based on the default scheduling, but an argument could also be made to default all these to the same timeout.  Basically, if the task isn't run in x minutes, no matter which scheduled task, expire it.

I looked into rate limits as well, but it's cumbersome to apply them and not very flexible. 

This doesn't change consume tasks, those will never expire and shouldn't ever be configured to do so.

Fixes #2604

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by shamoon — APPROVED** [review]
Thanks for sorting this out. ~~Just curious if you thought this was a better solution than https://docs.celeryq.dev/en/stable/userguide/workers.html#rate-limits ? Defer to you on all things celery, just curious~~

Nevermind, its not the same thing of course, now that I think about it

**Comment by shamoon** [comment]
Nevermind, its not the same thing of course, now that I think about it

**Comment by stumpylog** [comment]
I am having a look at rate limits, but they're a little weird, not so well documented and I'm not positive they are actually applying how I think yet.  So the simple change first.

**Comment by stumpylog** [comment]
@hashworks, if you'd like to chime in on the expiration settings?

File: src/paperless/settings.py [hunk 1]
```diff
@@ -117,27 +127,50 @@ def _parse_beat_schedule() -> Dict:
...
+            },
         },
         {
             "name": "Perform sanity check",
             "env_key": "PAPERLESS_SANITY_TASK_CRON",
             # Default Sunday at 00:30
             "env_default": "30 0 * * sun",
             "task": "documents.tasks.sanity_check",
+            "options": {
+                # 1 hour before default schedule sends again
```
Comment (hashworks) [inline src/paperless/settings.py:168]: That's not correct: This lets the sanity check expire 7 hours before the default schedule. Correct would be `(7.0*24.0-1.0)*60.0*60.0`

**Review by hashworks — COMMENTED** [review]
Sorry for the delay, I didn't find time to review it until now.

**Comment by hashworks** [comment]
Thanks!

**Pull request merged**